auth/roles-metadata.cc

```diff
@@ -51,54 +51,36 @@ future<bool> default_role_row_satisfies(
             meta::roles_table::qualified_name,
             meta::roles_table::role_col_name);
 
-    return do_with(std::move(p), [&qp](const auto& p) {
-        return qp.execute_internal(
-                query,
-                db::consistency_level::ONE,
-                {meta::DEFAULT_SUPERUSER_NAME},
-                cql3::query_processor::cache_internal::yes).then([&qp, &p](::shared_ptr<cql3::untyped_result_set> results) {
-            if (results->empty()) {
-                return qp.execute_internal(
-                        query,
-                        db::consistency_level::QUORUM,
-                        internal_distributed_query_state(),
-                        {meta::DEFAULT_SUPERUSER_NAME},
-                        cql3::query_processor::cache_internal::yes).then([&p](::shared_ptr<cql3::untyped_result_set> results) {
-                    if (results->empty()) {
-                        return make_ready_future<bool>(false);
-                    }
-
-                    return make_ready_future<bool>(p(results->one()));
-                });
-            }
-
-            return make_ready_future<bool>(p(results->one()));
-        });
-    });
+    for (auto c : { db::consistency_level::ONE, db::consistency_level::QUORUM }) {
```

Comment:
Unrelated: I think we should switch the order. First try quorum, then fall back to ONE on failure (not on empty rows).